sg_covid_impact/make_sic_division.py

```diff
@@ -1,128 +1,82 @@
+# %%
+"""Script to aggregate NOMIS data over SIC divisions."""
 import os
-import re
 import logging
+
 import pandas as pd
-from sg_covid_impact.getters.nomis import get_BRES, get_IDBR
-from sg_covid_impact.sic import load_sic_taxonomy, extract_sic_code_description
 
 import sg_covid_impact
+from sg_covid_impact.getters.nomis import get_BRES, get_IDBR
+from sg_covid_impact.sic import (
+    load_sic_taxonomy,
+    extract_sic_code_description,
+    save_sic_taxonomy,
+    _SIC_OUTPUT_FILE,
+)
+
 
 project_dir = sg_covid_impact.project_dir
 
 
-def make_sic_target_lookup(sic, source_level="Class", target_level="Division"):
-    """Makes a lookup between source SIC level (eg class ie SIC-4)
-        and a target level (eg Division)
-    Args:
-        sic (table): the sic taxonomy
-        source_level (str): the sic level we are converting from
-        target_level: the sic level we want to convert into
-    Returns:
-        A dict mapping source levels into target levels
-    """
-    lookup_table = (  # Each row in this lookup has a source and a target
-        sic[[target_level, source_level]].fillna(method="ffill").dropna(axis=0)
+def make_sic_div_lookup(sic):
```

Comment:
Fixes the bug flagged in last review.

For simplicity it restricts itself to create a division lookup (If we wanted something generic, we should focus the logic in a function that fills in all gaps in the spreadsheet)